src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/AbstractReporter.java

```diff
@@ -1,7 +1,80 @@
 package com.sap.sgs.phosphor.fosstars.tool.github;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sap.sgs.phosphor.fosstars.tool.Project;
 import com.sap.sgs.phosphor.fosstars.tool.Reporter;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-abstract class AbstractReporter implements Reporter {
+abstract class AbstractReporter<T extends Project> implements Reporter<T> {
+
+  /**
+   * For serialization.
+   */
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  /**
+   * A type reference of a list of {@link GitHubProject}s for deserialization.
+   */
+  private static final TypeReference<List<GitHubProject>> LIST_OF_GITHUB_PROJECTS_TYPE
+      = new TypeReference<List<GitHubProject>>() {};
+
+  /**
+   * Merges two lists of projects. The merge is based on URLs.
+   * If there are two projects with the same URLs,
+   * then the method uses the project from the first list.
+   *
+   * @param projects The first list of projects.
+   * @param extraProjects The second list of projects.
+   * @return A list of projects which contains all projects from the original lists.
+   */
+  static List<GitHubProject> merge(
+      List<GitHubProject> projects, List<GitHubProject> extraProjects) {
+
+    Map<URL, GitHubProject> map = new HashMap<>();
+    for (GitHubProject extraProject : extraProjects) {
+      map.put(extraProject.url(), extraProject);
+    }
+    for (GitHubProject project : projects) {
+      map.put(project.url(), project);
+    }
+
+    List<GitHubProject> allProjects = new ArrayList<>();
+    for (Map.Entry<URL, GitHubProject> entry : map.entrySet()) {
+      allProjects.add(entry.getValue());
+    }
+
+    return allProjects;
+  }
+
+  /**
+   * Loads projects from a JSON file.
+   * If the file doesn't exist, the the method returns an empty list.
+   *
+   * @param extraSourceFileName A path to the file.
+   * @return A list of loaded extra projects.
+   * @throws IOException If the projects couldn't be loaded.
+   */
+  static List<GitHubProject> loadProjects(String extraSourceFileName) throws IOException {
```

Comment:
Please consider this
- If the files are not found? Then we return an empty list.
- Why not catch the IOException also and then return the emptyList?

Reply:
The method tries to load a file only if a file name is specified (not null) and the file exists. However, if the file exists, the method makes sure that the content of the file is correct. That's why it doesn't return an empty list if deserialization throws an IOException.